File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.c
void setZeroes(int** matrix, int matrixSize, int* matrixColSize) {
    int isRowZero = 0, isColZero = 0;

    for (int j = 0; j < *matrixColSize; ++j) {
        if (matrix[0][j] == 0) {
            isRowZero = 1;
            break;
        }
    }

    for (int i = 0; i < matrixSize; ++i) {
        if (matrix[i][0] == 0) {
            isColZero = 1;
            break;
        }
    }

    for (int i = 1; i < matrixSize; ++i) {
        for (int j = 1; j < *matrixColSize; ++j) {
            if (matrix[i][j] == 0) {
                matrix[i][0] = 0;
                matrix[0][j] = 0;
            }
        }
    }

    for (int i = 1; i < matrixSize; ++i) {
        for (int j = 1; j < *matrixColSize; ++j) {
            if (matrix[i][0] == 0 || matrix[0][j] == 0) {
                matrix[i][j] = 0;
            }
        }
    }

    if (isRowZero) {
        for (int j = 0; j < *matrixColSize; ++j) {
            matrix[0][j] = 0;
        }
    }

    if (isColZero) {
        for (int i = 0; i < matrixSize; ++i) {
            matrix[i][0] = 0;
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
from typing import List


class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n, m = len(matrix), len(matrix[0])
        zero_sets = set(), set()

        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    zero_sets[0].add(i)
                    zero_sets[1].add(j)

        for i in range(n):
            for j in range(m):
                if i in zero_sets[0] or j in zero_sets[1]:
                    matrix[i][j] = 0
